scipy/sparse/dia.py

```diff
@@ -247,15 +247,28 @@ def transpose(self):
         data = data[r,c]
         return dia_matrix((data, offsets), shape=(num_cols,num_rows))
 
-    def tocsr(self, copy=False):
-        #this could be faster
-        return self.tocoo(copy=copy).tocsr()
+    def tocsc(self, copy=False):
+        from .csc import csc_matrix
+        if self.nnz == 0:
+            return csc_matrix(self.shape, dtype=self.dtype)
 
-    tocsr.__doc__ = spmatrix.tocsr.__doc__
+        num_rows, num_cols = self.shape
+        num_offsets, offset_len = self.data.shape
+        offset_inds = np.arange(offset_len)
 
-    def tocsc(self, copy=False):
-        #this could be faster
-        return self.tocoo(copy=copy).tocsc()
+        row = offset_inds - self.offsets[:,None]
+        mask = (row >= 0)
+        mask &= (row < num_rows)
+        mask &= (offset_inds < num_cols)
+        mask &= (self.data != 0)
+
+        indptr = np.zeros(num_cols + 1, dtype=row.dtype)
```

Comment:
This probably always creates 64-bit indices on 64-bit systems. (Not sure about win64, though...) Could be better to determine `idx_dtype` based on nnz.